src/utilities/index.ts

```diff
@@ -169,9 +169,10 @@ The function recursively merges the objects and returns the resulting object.
 */
 export const mergeErrors = <T extends FieldValues>(
   frontendErrors: Partial<FieldErrorsImpl<DeepRequired<T>>>,
-  backendErrors?: Partial<FieldErrorsImpl<DeepRequired<T>>>
+  backendErrors?: Partial<FieldErrorsImpl<DeepRequired<T>>>,
+  onMerge?: () => void
 ) => {
-  if (!backendErrors) {
+  if (!backendErrors || (onMerge && Object.keys(backendErrors).length === 0)) {
```

Comment:
Do you think the above change would block 2nd level object errors form being added?

eg. errors: { userName: { message: "Email is required", type: "custom" } }

This will not add.
```
  export const action = async ({ request }: ActionArgs) => {
const { errors, data } = await getValidatedFormData<FormData>(
    request,
    resolver
  );
  if (errors) {
    return json({ errors, ok: true });
  }
  ...
```

It has to be changed to the below, to log errors:
```
export const action = async ({ request }: ActionArgs) => {
  const { errors, data } = await getValidatedFormData<FormData>(
    request,
    resolver
  );
  if (errors) {
    return json({ ...errors, ok: true });
  }
 ...
```

Reply:
Needs some changes closing pr for now